engine/core/IgeEntity.js

```diff
@@ -103,13 +103,9 @@ var IgeEntity = IgeObject.extend({
 			this.streamUpdateData([{ isHidden: false }]);
 		} else if (ige.isClient) {
 			// this.disableInterpolation(false)
-
 			// add a little bit of delay before showing the item, so we don't see item translating from old location to new location
 			self._hidden = false;
-			var pixiEntity = self._pixiText || self._pixiTexture;
-			if (pixiEntity) {
-				pixiEntity.visible = true;
-			}
+			ige.client.emit('show', this);
```

Comment:
I think the idea behind this was we don't have a specific data class for like 'IgePixiEntity', so we have to send the entity as a parameter so that `IgePixi.js` knows what to do with the event.

If we were designing a new system that we would want to use in the future, I could think of better ways to do this, but those involve writing new classes that would contain these methods. Maybe I'm missing something...

Reply:
I think you are right. PIXI objects would have to keep track of the entity object etc. I was thinking in the terms of the way phaser classes work and this is direct change to master, so I need to think in the terms of old rendering logic, so your approach seems like a good balance.